data/sql/updates/pending_db_world/rev_1624655271618733500.sql

```diff
@@ -0,0 +1,5 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1624655271618733500');
+
+DELETE FROM `creature` WHERE (`id` = 23602) AND (`guid` IN (31046));
+INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES
+(31046, 23602, 1, 0, 0, 1, 1, 21640, 0, -3647.61, -4449.4, 15.008, 0.536, 360, 0, 0, 1009, 1067, 2, 0, 0, 0, '', 0);
```

Comment:
The line break should be added.
Otherwise, it will join with another line.

Reply:
Why use IN if it is a single record?

Reply:
Nice spot, that's how Keira exported it.

Reply:
> The line break should be added.
> Otherwise, it will join with another line.

I don't understand what you mean.

Reply:
We need a blank line at the end of any SQL file, because when we merge two files we can get the following result

file1.sql
```SQL
UPDATE something WHERE x = 2;
-- some random comment
```

file2.sql
```SQL
UPDATE something WHERE x = 2;
```

merged files:
```SQL
UPDATE something WHERE x = 2;
-- some random comment UPDATE something WHERE x = 2;
```

this can cause some issues when someone updates the sql data, because the `comment` will even disable the next line of sql for example, but it can also cause other kind of issues

Reply:
indeed this 
![image](https://user-images.githubusercontent.com/519778/123513192-03a76680-d68c-11eb-948f-0fe470e16bc7.png)
point "no new line at the end of file"

Reply:
Correct, as Helias told you in the previous comment. Always at the end of the document, there must be a line break, that is, a blank line. And rest to the SQL query, I would do it as follows.

```sql
DELETE FROM `creature` WHERE `id`=23602 AND `guid`=31046;
INSERT INTO `creature` (`guid`, `id`, `map`, `zoneId`, `areaId`, `spawnMask`, `phaseMask`, `modelid`, `equipment_id`, `position_x`, `position_y`, `position_z`, `orientation`, `spawntimesecs`, `wander_distance`, `currentwaypoint`, `curhealth`, `curmana`, `MovementType`, `npcflag`, `unit_flags`, `dynamicflags`, `ScriptName`, `VerifiedBuild`) VALUES
(31046, 23602, 1, 0, 0, 1, 1, 21640, 0, -3647.61, -4449.4, 15.008, 0.536, 360, 0, 0, 1009, 1067, 2, 0, 0, 0, '', 0);
```

In other words, if you are deleting / editing a single record / ID / GUID, instead of using IN, you can use equality. Because it is only 1. I hope it was understood this time.

The line break is solved in this way.

![line](https://user-images.githubusercontent.com/2810187/123517854-b6ba9480-d679-11eb-942e-dfd298e6ab7c.png)

Reply:
Thanks guys for the clarifications. 

I will take care of them.

This looks like a limitation of the sql migration tool we are using.

Reply:
using IN or = for single values doesn't really matter. Up to the author to decide, I vary between the two without thinking about it

Reply:
The idea was to mark possible errors for him to accommodate them. But yes, I guess each author can decide some things. For me in particular, if it is a single record, I prefer to use the =. On the other hand, if there are several, then I use the IN. But here the most important thing really, is the subject of the line break, so that it is not mixed with the content of another file.